guix: building for `aarch64-linux-gnu` on aarch64 fails

Running a Guix build of master (f002f8a0e76bd5abd60ef6809b1ff2b4c2d56071) using:
```bash
HOSTS="aarch64-linux-gnu" ./contrib/guix/guix-build
```
on aarch64 hardware (Alpine Docker container, running on Apple M1) currently fails with the following:
```bash
cc1: error: no include path in which to search for stdc-predef.h
/gnu/store/lzf5zg0diw2bhh2qji4bl2v46wd8pylc-bash-minimal-5.1.8/bin/bash ../../gcc-10.3.0/gcc/../move-if-change tmp-macro_list macro_list
echo timestamp > s-macro_list
rm -rf include-fixed; mkdir include-fixed
chmod a+rx include-fixed
if [ -d ../prev-gcc ]; then \
  cd ../prev-gcc && \
  make real-install-headers-tar DESTDIR=`pwd`/../gcc/ \
    libsubdir=. ; \
else \
  set -e; for ml in `cat fixinc_list`; do \
    sysroot_headers_suffix=`echo ${ml} | sed -e 's/;.*$//'`; \
    multi_dir=`echo ${ml} | sed -e 's/^[^;]*;//'`; \
    fix_dir=include-fixed${multi_dir}; \
    if ! false && test ! -d `echo /usr/include | sed -e :a -e 's,[^/]*/\.\.\/,,' -e ta`; then \
      echo The directory that should contain system headers does not exist: >&2 ; \
      echo "  `echo /usr/include | sed -e :a -e 's,[^/]*/\.\.\/,,' -e ta`" >&2 ; \
      tooldir_sysinc=`echo "/gnu/store/bp869rrz2z24yvjyma9kkj0alzxvqkc2-gcc-cross-sans-libc-aarch64-linux-gnu-10.3.0-lib/lib/gcc/aarch64-linux-gnu/10.3.0/../../../../../../../aarch64-linux-gnu/sys-include" | sed -e :a -e "s,[^/]*/\.\.\/,," -e ta`; \
      if test "x`echo /usr/include | sed -e :a -e 's,[^/]*/\.\.\/,,' -e ta`" = "x${tooldir_sysinc}"; \
      then sleep 1; else exit 1; fi; \
    fi; \
    /gnu/store/lzf5zg0diw2bhh2qji4bl2v46wd8pylc-bash-minimal-5.1.8/bin/bash ../../gcc-10.3.0/gcc/../mkinstalldirs ${fix_dir}; \
    chmod a+rx ${fix_dir} || true; \
    (TARGET_MACHINE='aarch64-unknown-linux-gnu'; srcdir=`cd ../../gcc-10.3.0/gcc; ${PWDCMD-pwd}`; \
      SHELL='/gnu/store/lzf5zg0diw2bhh2qji4bl2v46wd8pylc-bash-minimal-5.1.8/bin/bash'; MACRO_LIST=`${PWDCMD-pwd}`/macro_list ; \
      gcc_dir=`${PWDCMD-pwd}` ; \
      export TARGET_MACHINE srcdir SHELL MACRO_LIST && \
      cd ../build-aarch64-unknown-linux-gnu/fixincludes && \
      /gnu/store/lzf5zg0diw2bhh2qji4bl2v46wd8pylc-bash-minimal-5.1.8/bin/bash ./fixinc.sh "${gcc_dir}/${fix_dir}" \
        `echo /usr/include | sed -e :a -e 's,[^/]*/\.\.\/,,' -e ta`  ); \
    rm -f ${fix_dir}/syslimits.h; \
    if [ -f ${fix_dir}/limits.h ]; then \
      mv ${fix_dir}/limits.h ${fix_dir}/syslimits.h; \
    else \
      cp ../../gcc-10.3.0/gcc/gsyslimits.h ${fix_dir}/syslimits.h; \
    fi; \
    chmod a+r ${fix_dir}/syslimits.h; \
  done; \
fi
The directory that should contain system headers does not exist:
  /usr/include
make[3]: *** [Makefile:3172: stmp-fixinc] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-10.3.0.drv-0/build/gcc'
make[2]: *** [Makefile:4688: all-stage1-gcc] Error 2
make[2]: Leaving directory '/tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-10.3.0.drv-0/build'
make[1]: *** [Makefile:18381: stage1-bubble] Error 2
make[1]: Leaving directory '/tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-10.3.0.drv-0/build'
make: *** [Makefile:946: all] Error 2
```

[Full build log](https://gist.github.com/fanquake/df8d9a4f434d2d374cc3d2dd5492793d).

Tested with [the updated patch](https://github.com/fanquake/bitcoin/tree/test_22458_patch):
```bash
make[3]: Leaving directory '/tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-0/build/gcc'
mkdir -p -- aarch64-unknown-linux-gnu/libgcc
Checking multilib configuration for libgcc...
Configuring stage 1 in aarch64-unknown-linux-gnu/libgcc
configure: creating cache ./config.cache
checking build system type... aarch64-unknown-linux-gnu
checking host system type... aarch64-unknown-linux-gnu
checking for --enable-version-specific-runtime-libs... no
checking for a BSD-compatible install... /gnu/store/wiv74mq31s0vppzm99qwnjkd8xa3albw-coreutils-8.32/bin/install -c
checking for gawk... gawk
checking for aarch64-unknown-linux-gnu-ar... ar
checking for aarch64-unknown-linux-gnu-lipo... lipo
checking for aarch64-unknown-linux-gnu-nm... /tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-0/build/./gcc/nm
checking for aarch64-unknown-linux-gnu-ranlib... ranlib
checking for aarch64-unknown-linux-gnu-strip... strip
checking whether ln -s works... yes
checking for aarch64-unknown-linux-gnu-gcc... /tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-0/build/./gcc/xgcc -B/tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-0/build/./gcc/ -B/gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/bin/ -B/gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/lib/ -isystem /gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/include -isystem /gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/sys-include   
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-0/build/./gcc/xgcc -B/tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-0/build/./gcc/ -B/gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/bin/ -B/gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/lib/ -isystem /gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/include -isystem /gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/sys-include    accepts -g... yes
checking for /tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-0/build/./gcc/xgcc -B/tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-0/build/./gcc/ -B/gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/bin/ -B/gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/lib/ -isystem /gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/include -isystem /gnu/store/0x6x96gvifjp5zhcj48s9cf29zb0ffil-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0/aarch64-unknown-linux-gnu/sys-include    option to accept ISO C89... none needed
checking how to run the C preprocessor... /lib/cpp
configure: error: in `/tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-0/build/aarch64-unknown-linux-gnu/libgcc':
configure: error: C preprocessor "/lib/cpp" fails sanity check
See `config.log' for more details.
make[2]: *** [Makefile:16831: configure-stage1-target-libgcc] Error 1
make[2]: Leaving directory '/tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-0/build'
make[1]: *** [Makefile:21251: stage1-bubble] Error 2
make[1]: Leaving directory '/tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-0/build'
make: *** [Makefile:932: all] Error 2
command "make" "-j" "7" "inhibit_libc=true" "LDFLAGS_FOR_TARGET=-B/gnu/store/cb88z63hyg1icd2kkahiink2p291mhr2-glibc-2.31/lib -Wl,-dynamic-linker -Wl,/gnu/store/cb88z63hyg1icd2kkahiink2p291mhr2-glibc-2.31/lib/ld-linux-aarch64.so.1" "LDFLAGS=-Wl,-rpath=/gnu/store/cb88z63hyg1icd2kkahiink2p291mhr2-glibc-2.31/lib -Wl,-dynamic-linker -Wl,/gnu/store/cb88z63hyg1icd2kkahiink2p291mhr2-glibc-2.31/lib/ld-linux-aarch64.so.1" "BOOT_CFLAGS=-O2 -g0" failed with status 2
```

[Full log](https://gist.github.com/fanquake/774943a1aeed60f186ddae5cedeb4d44).

Then ran:
```bash
guix build /gnu/store/1rmh3vsc39ls8q5zqx9pn6613h5avpay-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv --keep-failed
```
The contents of that directory (`tmp/guix-build-gcc-cross-sans-libc-aarch64-linux-gnu-7.5.0.drv-6/`) is available here: https://github.com/fanquake/build_artifacts/releases/tag/guix_22458.

> FWIW, I've managed to build `gcc-cross-sans-libc-aarch64-linux-gnu-10.3.0` with [this](https://github.com/hebasto/bitcoin/commits/220522-guix.1) branch. pizza
> 
> Unfortunately, the following `glibc-cross-aarch64-linux-gnu-2.24` fails.
> 
> UPDATE: the full log is available [here](https://github.com/hebasto/artefacts/blob/master/guix-aarch64/0522-guix.1).

The error:
```
aarch64-linux-gnu-gcc   -D_RPC_THREAD_SAFE_ -D_GNU_SOURCE -DIS_IN_build -include /tmp/guix-build-glibc-cross-aarch64-linux-gnu-2.24.drv-0/build/config.h rpc_main.c \
	-o /tmp/guix-build-glibc-cross-aarch64-linux-gnu-2.24.drv-0/build/sunrpc/cross-rpc_main.o -MMD -MP -MF /tmp/guix-build-glibc-cross-aarch64-linux-gnu-2.24.drv-0/build/sunrpc/cross-rpc_main.o.dt -MT /tmp/guix-build-glibc-cross-aarch64-linux-gnu-2.24.drv-0/build/sunrpc/cross-rpc_main.o -c
rpc_main.c:37:10: fatal error: errno.h: No such file or directory
   37 | #include <errno.h>
      |          ^~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:171: /tmp/guix-build-glibc-cross-aarch64-linux-gnu-2.24.drv-0/build/sunrpc/cross-rpc_main.o] Error 1
```

looks weird because the `errno.h` header is available in many places. For example, `../source/include/errno.h`

> FWIW, I've managed to build `gcc-cross-sans-libc-aarch64-linux-gnu-10.3.0` with [this](https://github.com/hebasto/bitcoin/commits/220522-guix.1) branch. 🍕
> 
> Unfortunately, the following `glibc-cross-aarch64-linux-gnu-2.24` fails.
> 
> UPDATE: the full log is available [here](https://github.com/hebasto/artefacts/blob/master/guix-aarch64/0522-guix.1).

Another [step](https://github.com/hebasto/bitcoin/commits/220804-aarch64-guix) has been made. Now the following packages are being built successfully:
- `gcc-cross-sans-libc-aarch64-linux-gnu`
- `glibc-cross-aarch64-linux-gnu`

Building of `gcc-cross-aarch64-linux-gnu` fails with the following error:
```
/tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/./prev-gcc/xg++ -B/tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/./prev-gcc/ -B/gnu/store/x4r97fc5k2d2h44nh6gq1dyw9k85m7kx-gcc-cross-aarch64-linux-gnu-10.3.0/aarch64-linux-gnu/bin/ -nostdinc++ -B/tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/prev-aarch64-linux-gnu/libstdc++-v3/src/.libs -B/tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/prev-aarch64-linux-gnu/libstdc++-v3/libsupc++/.libs  -I/tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/prev-aarch64-linux-gnu/libstdc++-v3/include/aarch64-linux-gnu  -I/tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/prev-aarch64-linux-gnu/libstdc++-v3/include  -I/tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/gcc-10.3.0/libstdc++-v3/libsupc++ -L/tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/prev-aarch64-linux-gnu/libstdc++-v3/src/.libs -L/tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/prev-aarch64-linux-gnu/libstdc++-v3/libsupc++/.libs -c   -g -O2 -fno-checking -gtoggle -DIN_GCC     -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wno-error=format-diag -Wmissing-format-attribute -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings   -DHAVE_CONFIG_H  -DGENERATOR_FILE -fno-PIE -I. -Ibuild -I../../gcc-10.3.0/gcc -I../../gcc-10.3.0/gcc/build -I../../gcc-10.3.0/gcc/../include  -I../../gcc-10.3.0/gcc/../libcpp/include  \
	-o build/gengtype-parse.o ../../gcc-10.3.0/gcc/gengtype-parse.c
In file included from ./bconfig.h:3,
                 from ../../gcc-10.3.0/gcc/genchecksum.c:20:
./auto-host.h:2448:16: error: declaration does not declare anything [-fpermissive]
 2448 | #define rlim_t long
      |                ^~~~
In file included from ../../gcc-10.3.0/gcc/genchecksum.c:21:
../../gcc-10.3.0/gcc/system.h:488:14: error: conflicting declaration of C function ‘void* sbrk(int)’
  488 | extern void *sbrk (int);
      |              ^~~~
In file included from ../../gcc-10.3.0/gcc/system.h:294,
                 from ../../gcc-10.3.0/gcc/genchecksum.c:21:
/gnu/store/y8x9n30a8lb18qz9757phgjg6kmp0zzf-glibc-cross-aarch64-linux-gnu-2.32/include/unistd.h:1055:14: note: previous declaration ‘void* sbrk(intptr_t)’
 1055 | extern void *sbrk (intptr_t __delta) __THROW;
      |              ^~~~
In file included from ../../gcc-10.3.0/gcc/genchecksum.c:21:
../../gcc-10.3.0/gcc/system.h:496:14: error: ambiguating new declaration of ‘char* strstr(const char*, const char*)’
  496 | extern char *strstr (const char *, const char *);
      |              ^~~~~~
In file included from /tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/prev-aarch64-linux-gnu/libstdc++-v3/include/cstring:42,
                 from ../../gcc-10.3.0/gcc/system.h:235,
                 from ../../gcc-10.3.0/gcc/genchecksum.c:21:
/gnu/store/y8x9n30a8lb18qz9757phgjg6kmp0zzf-glibc-cross-aarch64-linux-gnu-2.32/include/string.h:327:1: note: old declaration ‘const char* strstr(const char*, const char*)’
  327 | strstr (const char *__haystack, const char *__needle) __THROW
      | ^~~~~~
In file included from ../../gcc-10.3.0/gcc/genchecksum.c:21:
../../gcc-10.3.0/gcc/system.h:544:20: error: conflicting declaration of C function ‘const char* strsignal(int)’
  544 | extern const char *strsignal (int);
      |                    ^~~~~~~~~
In file included from /tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/prev-aarch64-linux-gnu/libstdc++-v3/include/cstring:42,
                 from ../../gcc-10.3.0/gcc/system.h:235,
                 from ../../gcc-10.3.0/gcc/genchecksum.c:21:
/gnu/store/y8x9n30a8lb18qz9757phgjg6kmp0zzf-glibc-cross-aarch64-linux-gnu-2.32/include/string.h:462:14: note: previous declaration ‘char* strsignal(int)’
  462 | extern char *strsignal (int __sig) __THROW;
      |              ^~~~~~~~~
In file included from ../../gcc-10.3.0/gcc/system.h:695,
                 from ../../gcc-10.3.0/gcc/genchecksum.c:21:
../../gcc-10.3.0/gcc/../include/libiberty.h:112:14: error: ambiguating new declaration of ‘char* basename(const char*)’
  112 | extern char *basename (const char *) ATTRIBUTE_RETURNS_NONNULL ATTRIBUTE_NONNULL(1);
      |              ^~~~~~~~
In file included from /tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/prev-aarch64-linux-gnu/libstdc++-v3/include/cstring:42,
                 from ../../gcc-10.3.0/gcc/system.h:235,
                 from ../../gcc-10.3.0/gcc/genchecksum.c:21:
/gnu/store/y8x9n30a8lb18qz9757phgjg6kmp0zzf-glibc-cross-aarch64-linux-gnu-2.32/include/string.h:508:26: note: old declaration ‘const char* basename(const char*)’
  508 | extern "C++" const char *basename (const char *__filename)
      |                          ^~~~~~~~
In file included from ./bconfig.h:3,
                 from ../../gcc-10.3.0/gcc/genhooks.c:20:
./auto-host.h:2448:16: error: declaration does not declare anything [-fpermissive]
 2448 | #define rlim_t long
      |                ^~~~
In file included from ../../gcc-10.3.0/gcc/genhooks.c:21:
../../gcc-10.3.0/gcc/system.h:488:14: error: conflicting declaration of C function ‘void* sbrk(int)’
  488 | extern void *sbrk (int);
      |              ^~~~
In file included from ../../gcc-10.3.0/gcc/system.h:294,
                 from ../../gcc-10.3.0/gcc/genhooks.c:21:
/gnu/store/y8x9n30a8lb18qz9757phgjg6kmp0zzf-glibc-cross-aarch64-linux-gnu-2.32/include/unistd.h:1055:14: note: previous declaration ‘void* sbrk(intptr_t)’
 1055 | extern void *sbrk (intptr_t __delta) __THROW;
      |              ^~~~
In file included from ../../gcc-10.3.0/gcc/genhooks.c:21:
../../gcc-10.3.0/gcc/system.h:496:14: error: ambiguating new declaration of ‘char* strstr(const char*, const char*)’
  496 | extern char *strstr (const char *, const char *);
      |              ^~~~~~
In file included from /tmp/guix-build-gcc-cross-aarch64-linux-gnu-10.3.0.drv-0/build/prev-aarch64-linux-gnu/libstdc++-v3/include/cstring:42,
                 from ../../gcc-10.3.0/gcc/system.h:235,
                 from ../../gcc-10.3.0/gcc/genhooks.c:21:
/gnu/store/y8x9n30a8lb18qz9757phgjg6kmp0zzf-glibc-cross-aarch64-linux-gnu-2.32/include/string.h:327:1: note: old declaration ‘const char* strstr(const char*, const char*)’
  327 | strstr (const char *__haystack, const char *__needle) __THROW
      | ^~~~~~
make[3]: *** [Makefile:2723: build/genchecksum.o] Error 1
```

--- a/contrib/guix/manifest.scm
+++ b/contrib/guix/manifest.scm
@@ -78,6 +78,11 @@ http://www.linuxfromscratch.org/hlfs/view/development/chapter05/gcc-pass1.html"
                  (("-rpath=") "-rpath-link="))
                #t))))))))
 
+(define building-on (string-append (list-ref (string-split (%current-system) #\-) 0) "-guix-linux-gnu"))
+
+(define (explicit-cross-configure package)
+  (package-with-extra-configure-variable package "--build" building-on))
+
 (define (make-cross-toolchain target
                               base-gcc-for-libc
                               base-kernel-headers
@@ -87,9 +92,9 @@ http://www.linuxfromscratch.org/hlfs/view/development/chapter05/gcc-pass1.html"
   (let* ((xbinutils (cross-binutils target))
          ;; 1. Build a cross-compiling gcc without targeting any libc, derived
          ;; from BASE-GCC-FOR-LIBC
-         (xgcc-sans-libc (cross-gcc target
-                                    #:xgcc base-gcc-for-libc
-                                    #:xbinutils xbinutils))
+         (xgcc-sans-libc (explicit-cross-configure (cross-gcc target
+                                                              #:xgcc base-gcc-for-libc
+                                                              #:xbinutils xbinutils)))
          ;; 2. Build cross-compiled kernel headers with XGCC-SANS-LIBC, derived
          ;; from BASE-KERNEL-HEADERS
          (xkernel (cross-kernel-headers target
@@ -98,17 +103,17 @@ http://www.linuxfromscratch.org/hlfs/view/development/chapter05/gcc-pass1.html"
                                         xbinutils))
          ;; 3. Build a cross-compiled libc with XGCC-SANS-LIBC and XKERNEL,
          ;; derived from BASE-LIBC
-         (xlibc (cross-libc target
-                            base-libc
-                            xgcc-sans-libc
-                            xbinutils
-                            xkernel))
+         (xlibc (explicit-cross-configure (cross-libc target
+                                                      base-libc
+                                                      xgcc-sans-libc
+                                                      xbinutils
+                                                      xkernel)))
          ;; 4. Build a cross-compiling gcc targeting XLIBC, derived from
          ;; BASE-GCC
-         (xgcc (cross-gcc target
-                          #:xgcc base-gcc
-                          #:xbinutils xbinutils
-                          #:libc xlibc)))
+         (xgcc (explicit-cross-configure (cross-gcc target
+                                                    #:xgcc base-gcc
+                                                    #:xbinutils xbinutils
+                                                    #:libc xlibc))))
     ;; Define a meta-package that propagates the resulting XBINUTILS, XLIBC, and
     ;; XGCC
     (package

